pkg/profiling/fifo_dir.go

```diff
@@ -0,0 +1,73 @@
+package profiling
+
+import (
+	"io/fs"
+	"os"
+	"path"
+	"sort"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	fifoDefaultMaxFileCount = 10
+)
+
+type fifoDir struct {
+	maxFileCount int
+	dirPath      string
+}
+
+func (fd fifoDir) Create(fileName string) (*os.File, error) {
+	err := os.MkdirAll(fd.dirPath, os.ModePerm)
+	if err != nil {
+		if !os.IsExist(err) {
+			return nil, errors.Wrapf(err, "creating directory: %s", fd.dirPath)
+		}
+	}
+
+	entries, err := os.ReadDir(fd.dirPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "reading directory: %s", fd.dirPath)
+	}
+
+	entryInfos, err := dirEntriesToFileInfo(entries)
+	if err != nil {
+		return nil, err
+	}
+
+	sort.Slice(entryInfos, func(i, j int) bool {
+		return entryInfos[i].ModTime().Before(entryInfos[j].ModTime())
+	})
+
+	for len(entryInfos) >= fd.maxFileCount {
+		rmPath := path.Join(fd.dirPath, entryInfos[0].Name())
+		err := os.Remove(rmPath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "removing file: %s", rmPath)
+		}
+		entryInfos = entryInfos[1:]
```

Comment:
Wouldn't it be more straightforward to compute the "to delete" sub-slice in one go, and the just iterate over it, rather than do this re-slicing? :thinking:

Reply:
It does to me, but meh :shrug: